### DISCUSSION TITLE
Power on / off a esp 8266 wifi relai with linux command line

### DISCUSSION THREAD

**Discussion by GregESP19** [arendst/Tasmota#16320] [Q&A] [answered]
Hello,
I need to find a command under linux for activate / desactivate my wifi relai
Under firefox this URL work fine :

for activate :
http://192.168.1.120/cm?cmnd=Power%20On

for desactivate :
http://192.168.1.120/cm?cmnd=Power%20Off

I try this command under the terminal on linux :

echo -en "Power%20On" | nc -w 0 192.168.1.120 80
or
echo -en "cm?cmnd=Power%20On" | nc -w 0 192.168.1.120 80

But it doesn't work

Thanks for your help!

**Comment by sfromis** [comment]
Using MQTT is recommended, where you can use tools like `mosquitto_sub`

Whatever happens with your use of the nc command is likely you having to work out how to use this tool, not a Tasmota issue.

Using curl to do same worked when I did a small test.

According to a very cursory Google search, what you pass to nc is the full http request, not merely parameters to match the IP you use.

**Reply by GregESP19** [reply]
Hello sfromis,
Thanks for your answer. I'm going to learn how MQTT works. Thanks this tool i could use all the tasmota commands.

**Comment by lalo-uy** [comment, marked as answer]
wget -q -T2 -t2 -O - http://your_ip/cm?cmnd=Power%20On;

**Reply by GregESP19** [reply]
Hello lalo-uy,
I had tested your command line. It's WORKS FINE!!!!!
I have earn a lot of time!!
Thanks you so munch

**Comment by GregESP19** [comment]
[](url
![IMG_20220824_095058_resize_29](https://user-images.githubusercontent.com/111926581/186364078-297e1220-8372-4261-a518-9dbb6b728c13.jpg)
)

**Reply by sfromis** [reply]
Don't expect much in battery life. ESP8266 with Wifi is not very good for battery operation.

**Reply by GregESP19** [reply]
Hello,
It is a test : i want to kwow how many time it's works with au 19800 mAh battery (model 18650 Li-Ion rechargeable)
I will give you some news
Thanks

**Reply by sfromis** [reply]
Li-ion 18650 with a claim of 19800 mAh capacity sounds like a bad fake.

**Reply by GregESP19** [reply]
Aliexpress ;-))
But you've reason : battery is discharged in 4 hours

![IMG_20220824_122713_resize_35](https://user-images.githubusercontent.com/111926581/186396430-b209f48b-36b9-423f-a2ab-101fd6aeecf3.jpg)
s

**Reply by sfromis** [reply]
Maybe the fakery is just "forgetting" a decimal point before the last two digits.

**Reply by barbudor** [reply]
My understanding is tha the best LiIon batteries have an energy density of 265 Wh/kg, with a cell at 3.7V this means 71Ah/kg
So your battery should weight at least 3.8 kg to provide such energy

**Reply by GregESP19** [reply]
![e8207cb6-71f9-11eb-a88b-02fe89184577](https://user-images.githubusercontent.com/111926581/186403527-5bda3612-51fc-4455-9af1-0e906cb63510.jpg)

**Reply by sfromis** [reply]
Those numbers look "off". 19.8 A at 3.7V for one hour would be 73.26 Wh, meaning that 3.6 of those batteries would be 265 Wh. With "only" 73 Wh, that would be a weight of near 276g.

Known good Li-ion 18650 cells have a capacity of something like 3100 mAh and a weight around 46g. That's 11.5 Wh, or indeed almost 250 Wh/kg including the can and wrapping.

No reason to paint them as bigger liars than they really are!

**Reply by sfromis** [reply]
BTW. The board you use there looks strongly like it is meant for 5V, not a much lower voltage. And those relay boards with an ESP-01 doing serial communication with the actually controlling MCU are IMO so much worse than a board with straight pins from the ESP8266, that I'd not recommend buying such a mess.

**Reply by GregESP19** [reply]
Usualy i use 12V to 5V converter (aliexpress too but works fine since many years)


![Détail Boitier Alerte](https://user-images.githubusercontent.com/111926581/186406948-bb9b5b28-4452-46ab-8e78-8e5b57986d80.jpg)

**Reply by barbudor** [reply]
Indeed I did the operation in the wrong side

@GregESP19 Tasmota is part of the new tactic of Gendarmerie? 😂👍

So what's the purpose of the whole stuff with the NFC tag ?

**Reply by sfromis** [reply]
You can also get relay boards powered by 12V without adding a conveter.

**Reply by GregESP19** [reply]
Have you somes links for relay wifi 12V? I'm interesting.
And yes, this is a project for Gendarmerie. Relay are coupled with a SMS server i'had created. NFC tag are progammed to send a sms to the server when you scan it with a gsm. So sms with preformated message is disaptched to all people and relay wifi turn on a sound siren and the red light. Server is all in bash so i need the command to trigger the relay

**Reply by GregESP19** [reply]
Actually i don't use tasmota with my system because i prefere to work in AP mode for security reason.
But i'm interested by tasmota for my own.

**Reply by sfromis** [reply]
Many relay boards in the templates repository https://templates.blakadder.com/relay.html
The description there does not tell which ones supports 12V, but many do.
